Fleece/Integration/MDict.hh

```diff
@@ -137,8 +138,9 @@ namespace fleece {
                 return true;
             MCollection::mutate();
             _map.clear();
+            _newKeys.clear();
             for (Dict::iterator i(_dict); i; ++i)
-                _map.emplace(i.keyString(), MValue::empty);
```

Comment:
This change isn't necessary. `i.keyString()` is owned by `_dict` so we don't need to make our own copy of it.

Reply:
I try to maintain the invariant that: all keys, as slices, of _map are allocated in alloc_slices held in _newKeys. And I am afraid this is actually the critical part of this commit. I'll double check it.